src/cc/BPF.h

```diff
@@ -111,31 +106,31 @@ class BPF {
   StatusTuple unload_func(const std::string& func_name);
 
   std::string get_kprobe_event(const std::string& kernel_func,
-                               bpf_attach_type type);
+                               int attach_type);
   std::string get_uprobe_event(const std::string& binary_path, uint64_t offset,
-                               bpf_attach_type type);
+                               int attach_type);
 
   StatusTuple detach_kprobe_event(const std::string& event, open_probe_t& attr);
   StatusTuple detach_uprobe_event(const std::string& event, open_probe_t& attr);
   StatusTuple detach_tracepoint_event(const std::string& tracepoint,
                                       open_probe_t& attr);
   StatusTuple detach_perf_event_all_cpu(open_probe_t& attr);
 
-  std::string attach_type_debug(bpf_attach_type type) {
-    switch (type) {
-    case bpf_attach_type::probe_entry:
+  std::string attach_type_debug(int attach_type) {
+    switch (attach_type) {
```

Comment:
for example, by keeping enum compiler will be able to warn here that some enum values are not handled in the switch...